File: tests/fixtures/negative/security/WeakMacAlgorithm.java
package test;

class Crypto {
    void mac() throws Exception {
        javax.crypto.Mac.getInstance("HmacSHA256");
        javax.crypto.Mac.getInstance("HmacSHA384");
        javax.crypto.Mac.getInstance("HmacSHA512");
        Mac.getInstance("HmacSHA1");
    }
}

class Mac {
    static Mac getInstance(String name) {
        return new Mac();
    }
}

File: tests/fixtures/negative/security/WeakMacAlgorithm.kt
package test

class Crypto {
    fun mac() {
        javax.crypto.Mac.getInstance("HmacSHA256")
        javax.crypto.Mac.getInstance("HmacSHA384")
        javax.crypto.Mac.getInstance("HmacSHA512")
        Mac.getInstance("HmacSHA1")
    }
}

class Mac {
    companion object {
        fun getInstance(name: String): Mac = Mac()
    }
}
